utility/doc/README.md

```diff
@@ -156,6 +156,67 @@ GetProposalTransactions(proposer []byte, maxTransactionBytes int, lastBlockByzan
 ApplyBlock(Height int64, proposer []byte, transactions [][]byte, lastBlockByzantineValidators [][]byte) (appHash []byte, err error)
 ```
 
+## Protocols
+
+##### Session Protocol
+`Pocket` implements the V1 Utility Specification's Session Protocol by satisfying the following interface
+```golang
+type Session interface {
+	NewSession(sessionHeight int64, blockHash string, geoZone GeoZone, relayChain RelayChain, application modules.Actor) (Session, types.Error)
+	GetServiceNodes() []modules.Actor // the ServiceNodes providing Web3 to the application
+	GetFishermen() []modules.Actor    // the Fishermen monitoring the serviceNodes
+	GetApplication() modules.Actor    // the Application consuming the web3 access
+	GetRelayChain() RelayChain        // the chain identifier of the web3
+	GetGeoZone() GeoZone              // the geolocation zone where all are registered
+	GetSessionHeight() int64          // the block height when the session started
+}
+```
+
+The general flow is:
+1) Create a session object from the seed data
+2) Create a key concatenating and hashing the seed data
+    - `sessionHeight + blockHash + geoZone + relayChain + appPublicKey`
+3) Get an ordered list of the public keys of serviceNodes who are:
+    - actively staked
+    - staked within geo-zone
+    - staked for relay-chain
+4) Pseudo-insert the session `key` string into the list and find the first actor directly below on the list
+5) Determine a new seedKey with the following formula: `Hash( key + actor1PublicKey )`
+6) Repeat steps 5 and 6 until all N serviceNodes are found
+7) Do steps 3 - 6 for Fishermen as well
+
+FAQ:
+- Q) why do we hash to find a newKey between every actor selection?
+- A) pseudo-random selection only works if each iteration is re-randomized or it would be subject to lexicographical proximity bias attacks
+
+Session Flow
+```mermaid
+sequenceDiagram
+    autonumber
+    participant WorldState
```

Comment:
I searched within the codebase and I found no further references to `WorldState`.

Is the `WorldState` a definition that we are going to reference and use going forward? If so, perhaps a 2 liner description could be beneficial **if it doesn't sound overkill/too obvious already**.

Maybe here it's pretty clear but if used in a different context one might not know/consider the fact that the height is an important factor.

I wouldn't do anything more than this, it could find home in the convenient FAQ section above:

> `WorldState` represents a queryable view on the internal state of the network at a certain height.